### ISSUE TITLE
best result on nuscenes

### ISSUE THREAD

**Issue by TianhaoFu** [tianweiy/CenterPoint#260]
hi,
i noticed that your rank in nuscenes lidar track is 2
```
CenterPoint-VID
mAP: 0.674
NDS: 0.718
```
What config should I use to train the model to achieve this effect?

**Comment by tianweiy** [comment]
this one is not my entry

**Comment by TianhaoFu** [comment]
thanks.

Which ranking on the nuscene list is yours now? Can you share the training strategy and related configuration?

@tianweiy

**Comment by tianweiy** [comment]
the leaderboard uses old version (and worse). Currently we get 66.8 / 68.5 (single / flip) https://github.com/tianweiy/CenterPoint/issues/249 on val set. The test set should be similar or higher. The configs are already provided. 

If you want to get to SOTA performance, you will need fusion based methods

**Comment by TianhaoFu** [comment]
ok, thanks!

**Comment by TianhaoFu** [comment]
> the leaderboard uses old version (and worse). The most recent models get 66.8 / 68.5 (single / flip) #249 on val set. The test set should be similar or higher. The configs are already provided.
> 
> If you want to get to SOTA performance, you will need fusion based methods (e.g. https://github.com/tianweiy/MVP)

Hi, i want to know does the result of 68.5 use dcn?
thanks!

**Comment by TianhaoFu** [comment]
> > the leaderboard uses old version (and worse). The most recent models get 66.8 / 68.5 (single / flip) #249 on val set. The test set should be similar or higher. The configs are already provided.
> > If you want to get to SOTA performance, you will need fusion based methods (e.g. https://github.com/tianweiy/MVP)
> 
> Hi, i want to know does the result of 68.5 use dcn? thanks!
@tianweiy

**Comment by tianweiy** [comment]
No, I don't think dcn helps much

**Comment by tianweiy** [comment]
I basically just run this config with pretrained model https://github.com/tianweiy/CenterPoint/blob/master/configs/mvp/nusc_centerpoint_voxelnet_0075voxel_fix_bn_z_scale.py

**Comment by TianhaoFu** [comment]
> I basically just run this config (w/ w/o flip test) with pretrained model https://github.com/tianweiy/CenterPoint/blob/master/configs/mvp/nusc_centerpoint_voxelnet_0075voxel_fix_bn_z_scale.py

can you provide the per-trained model? thanks!

**Comment by tianweiy** [comment]
i just use this model https://github.com/tianweiy/CenterPoint/blob/master/configs/nusc/README.md (first one)

**Comment by TianhaoFu** [comment]
so you trained totally 40 epoch?
could you provide your training command? like
```
python -m torch.distributed.launch --nnodes=1 ./tools/train.py configs/mvp/nusc_centerpoint_voxelnet_0075voxel_fix_bn_z_scale.py --resume_from epoch_20.pth
```
thanks!

**Comment by tianweiy** [comment]
i don't get it. I only trained 20 epochs, I didn't change anything (the config is exactly what i used)

**Comment by tianweiy** [comment]
run config < i mean testing

**Comment by tianweiy** [comment]
the 68.5 is with flip aug, 66.8 is without flip aug. Flip definitely helps at the cost of 4x latency

**Comment by TianhaoFu** [comment]
does the flip mean double flip?
should i add the double_flip=ture during training and testing

**Comment by tianweiy** [comment]
we always do flip aug during training.

By default, we don't do flip aug during testing.  

To enable it, we need to add multiple flip flags in the config. See this file for an example https://github.com/tianweiy/CenterPoint/blob/master/configs/nusc/voxelnet/nusc_centerpoint_voxelnet_0075voxel_dcn_flip.py

**Comment by TianhaoFu** [comment]
does the flip aug during testing help improve performance?

**Comment by tianweiy** [comment]
yes 

> the 68.5 is with flip aug, 66.8 is without flip aug.

**Comment by TianhaoFu** [comment]
> the 68.5 is with flip aug, 66.8 is without flip aug. Flip definitely helps at the cost of 4x latency

could you provide this test config with flip aug? 
thanks!

**Comment by tianweiy** [comment]
will do in a bit (after i finish graphics hw, probably 1 to 2 hrs)

**Comment by TianhaoFu** [comment]
i noticed that in https://github.com/tianweiy/CenterPoint/blob/master/configs/mvp/nusc_centerpoint_voxelnet_0075voxel_fix_bn_z_scale.py config. 
there is no voxel_generator config, why?

**Comment by tianweiy** [comment]
> there is no voxel_generator config, why?

it uses dynamic voxelization, so we can remove the old voxelizer 

> there is no difference, we use separate files for diff paper

**Comment by tianweiy** [comment]
> will do in a bit (after i finish graphics hw, probably 1 to 2 hrs)

up now
